samples/ipc_call_interface/producers/cipher_v1i1/interfaces/cipher_v1i1_interface.c

```diff
@@ -23,58 +23,124 @@
 #include <stdlib.h>
 #include <string.h>
 
-/*
- * Concrete implementations of the cipher commands.
- */
 static az_result cipher_1_encrypt_concrete(az_ulib_model_in model_in, az_ulib_model_out model_out)
 {
-  az_result result;
-
-  /*
-   * ==================
-   * The user code starts here.
-   */
   const cipher_1_encrypt_model_in* const in = (const cipher_1_encrypt_model_in* const)model_in;
   cipher_1_encrypt_model_out* out = (cipher_1_encrypt_model_out*)model_out;
+  return cipher_v1i1_encrypt(in->context, in->src, out->dest);
+}
 
-  result = cipher_v1i1_encrypt(
-      in->context, in->src, in->src_size, in->dst_buffer_size, out->dst, out->dst_size);
-
-  /*
-   * The user code ends here.
-   * ==================
-   */
-
-  return result;
+static az_result cipher_1_encrypt_span_wrapper(az_span model_in_span, az_span* model_out_span)
+{
+  AZ_ULIB_TRY
+  {
+    // Unmarshalling JSON in model_in_span to encrypt_model_in.
+    az_json_reader jr;
+    cipher_1_encrypt_model_in encrypt_model_in = { 0 };
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_init(&jr, model_in_span, NULL));
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+    {
+      if (az_json_token_is_text_equal(&jr.token, AZ_SPAN_FROM_STR(CIPHER_1_ENCRYPT_CONTEXT_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_token_get_uint32(&jr.token, &encrypt_model_in.context));
+      }
+      else if (az_json_token_is_text_equal(&jr.token, AZ_SPAN_FROM_STR(CIPHER_1_ENCRYPT_SRC_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+        encrypt_model_in.src
+            = az_span_create(az_span_ptr(jr.token.slice), az_span_size(jr.token.slice));
+      }
+      AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    }
+    AZ_ULIB_THROW_IF_AZ_ERROR(AZ_ULIB_TRY_RESULT);
+
+    // Create a temporary buffer to store the encrypt_model_out.
+    char dest_buffer[200];
```

Comment:
can you use a #define for the size here?

Reply:
I'll actually use model_out_span as buffer instead of create a new one in the stack. 

I'll do the same for decript as well.